packages/react-components/react-charts-preview/src/components/Sparkline/Sparkline.tsx

```diff
@@ -1,16 +1,146 @@
+    const [height, setHight] = React.useState<number>(20);
+    const [valueTextWidth, setValueTextWidth] = React.useState<number>(80);
+
+    const line = React.useMemo(() => {
+      return (
+        d3Line()
+          /* eslint-disable @typescript-eslint/no-explicit-any */
+          .x((d: any) => x(d.x))
+          /* eslint-disable @typescript-eslint/no-explicit-any */
+          .y((d: any) => y(d.y))
+          .curve(d3curveLinear)
+      );
+    }, [x, y]);
+
+    const area = React.useMemo(() => {
+      return (
+        d3Area()
+          /* eslint-disable @typescript-eslint/no-explicit-any */
+          .x((d: any) => x(d.x))
+          .y0(height)
+          /* eslint-disable @typescript-eslint/no-explicit-any */
+          .y1((d: any) => y(d.y))
+          .curve(d3curveLinear)
+      );
+    }, [height, x, y]);
+
+    React.useEffect(() => {
+      if (!_isChartEmpty()) {
+        // eslint-disable-next-line @typescript-eslint/no-shadow
+        const _points = props.data!.lineChartData![0].data;
+
+        /* eslint-disable @typescript-eslint/no-explicit-any */
+        const [xMin, xMax] = d3Extent(_points, (d: any) => d.x);
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        x = d3ScaleLinear()
+          .domain([xMin, xMax])
+          .range([margin.left!, width - margin.right!]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        y = d3ScaleLinear()
+          /* eslint-disable @typescript-eslint/no-explicit-any */
+          .domain([0, d3Max(_points, (d: any) => d.y)])
+          .range([height - margin.bottom!, margin.top!]);
+
+        setPoints(_points);
+      }
+    }, []);
+
+    const { data } = props;
+    function _isChartEmpty(): boolean {
+      return !(
+        data &&
+        data.lineChartData &&
+        data.lineChartData.length > 0 &&
+        data.lineChartData.filter(item => item.data.length === 0).length === 0
+      );
+    }
+    function drawSparkline() {
+      if (!line || !area) {
+        return null;
+      }
+      return (
+        <>
+          <path
+            className="line"
+            d={line(points)}
+            fill={'transparent'}
+            opacity={1}
+            strokeWidth={2}
+            stroke={data!.lineChartData![0].color!}
+          />
+          <path
+            className="area"
+            d={area(points)}
+            opacity={1}
+            fillOpacity={0.2}
+            fill={data!.lineChartData![0].color!}
+            role="img"
+            aria-hidden
+          />
+        </>
+      );
+    }
+
+    const classes = useSparklineStyles_unstable(props);
+    const focusAttributes = useFocusableGroup();
+    return !_isChartEmpty() ? (
+      <div className={classes.inlineBlock} {...focusAttributes}>
+        {width >= 50 && height >= 16 ? (
+          <svg
+            width={width}
+            height={height}
+            data-is-focusable={true}
```

Comment:
remove data is focusable attribute. It is no longer needed

Reply:
Okay